manifest/morph/morph.go

```diff
@@ -710,3 +766,17 @@ func attributePathSummary(p *tftypes.AttributePath) string {
 	}
 	return b.String()
 }
+
+func ValidateValue(t tftypes.Type, val interface{}, p *tftypes.AttributePath) []*tfprotov5.Diagnostic {
```

Comment:
We don't need to export this function so it can be called `validateValue()`

Reply:
We could maybe go a step further here and have a function called `newValue()` that wraps the calls to `ValidateValue()` and `NewValue()` and use that at every call site that exists now for `NewValue()`.